### ISSUE TITLE
Seeing error messages when setting multiple evaluation metrics

### ISSUE THREAD

**Issue by XiaoxiaoWang87** [dmlc/xgboost#1227]
I'm using the python version of Xgboost. When I try to set both auc and logloss as my evaluation metrics as follows:

```
param = {
         'bst:max_depth':4,
         'bst:eta':0.1, 
         'silent':1,
         'objective':'binary:logistic'
         }

param['nthread'] = 10
param['eval_metric'] = ["auc", "logloss"]
param['seed'] = 0

plst = param.items()

evallist  = [(dtrain_test1,'train'), (dtest2,'eval')]

num_round = 50
bst = xgb.train( plst, dtrain_test1, num_round, evallist, early_stopping_rounds = 5)
```

I got the error message: 

```
[05:51:24] dmlc-core/include/dmlc/./logging.h:245: [05:51:24] src/metric/metric.cc:21: Unknown metric function ['auc', 'logloss']
Traceback (most recent call last):
  File "model_bin.py", line 88, in <module>
    main()
  File "model_bin.py", line 81, in main
    bst = xgb.train( plst, dtrain_test1, num_round, evallist, early_stopping_rounds = 5)
  File "/home/xwang/anaconda3/lib/python3.5/site-packages/xgboost-0.4-py3.5.egg/xgboost/training.py", line 179, in train
    bst.update(dtrain, i, obj)
  File "/home/xwang/anaconda3/lib/python3.5/site-packages/xgboost-0.4-py3.5.egg/xgboost/core.py", line 780, in update
    _check_call(_LIB.XGBoosterUpdateOneIter(self.handle, iteration, dtrain.handle))
  File "/home/xwang/anaconda3/lib/python3.5/site-packages/xgboost-0.4-py3.5.egg/xgboost/core.py", line 101, in _check_call
    raise XGBoostError(_LIB.XGBGetLastError())
xgboost.core.XGBoostError: b"[05:51:24] src/metric/metric.cc:21: Unknown metric function ['auc', 'logloss']"

```

If I only set one evaluation metric: e.g.
    param['eval_metric'] = "auc"

everything works fine. What is the cause of the problem?

**Comment by khotilov** [comment]
Hmm... I couldn't reproduce your error.
Could you please try with the most recent code, and also include my multi-metric fix in #1239 .

**Comment by sisiruowan** [comment]
I've met this problem too, I suspect it's caused by string  adaptive. My code as follows(runs in Jupyter, python 2.7):
```
from __future__ import print_function
from __future__ import unicode_literals

xgb_param = {'base_score': 0.5,
 'booster': 'gbtree',
 'colsample_bylevel': 1,
 'colsample_bytree': 0.8,
 'gamma': 0,
 'learning_rate': 0.1,
 'max_delta_step': 0,
 'max_depth': 5,
 'min_child_weight': 1,
 'missing': None,
 'n_estimators': 1000,
 'nthread': 4,
 'objective': 'multi:softprob',
 'reg_alpha': 0,
 'reg_lambda': 1,
 'scale_pos_weight': 1,
 'seed': 27,
 'silent': 1,
 'subsample': 0.8,
 'num_class':11}
xgtrain = xgb.DMatrix(train_X, label=train_y)
cvresult = xgb.cv(xgb_param, xgtrain, num_boost_round=xgb_param['n_estimators'], nfold=5,
                          metrics='merror', early_stopping_rounds=50)
```
I got the error message:
```
XGBoostErrorTraceback (most recent call last)
<ipython-input-4-a4d15e73f9c5> in <module>()
     21 xgtrain = xgb.DMatrix(train_X, label=train_y)
     22 cvresult = xgb.cv(xgb_param, xgtrain, num_boost_round=xgb_param['n_estimators'], nfold=5,
---> 23                           metrics='merror', early_stopping_rounds=50)

/home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/training.pyc in cv(params, dtrain, num_boost_round, nfold, stratified, folds, metrics, obj, feval, maximize, early_stopping_rounds, fpreproc, as_pandas, verbose_eval, show_stdv, seed, callbacks, shuffle)
    404                            evaluation_result_list=None))
    405         for fold in cvfolds:
--> 406             fold.update(i, obj)
    407         res = aggcv([f.eval(i, feval) for f in cvfolds])
    408 

/home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/training.pyc in update(self, iteration, fobj)
    216     def update(self, iteration, fobj):
    217         """"Update the boosters for one iteration"""
--> 218         self.bst.update(self.dtrain, iteration, fobj)
    219 
    220     def eval(self, iteration, feval):

/home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/core.pyc in update(self, dtrain, iteration, fobj)
    893         if fobj is None:
    894             _check_call(_LIB.XGBoosterUpdateOneIter(self.handle, ctypes.c_int(iteration),
--> 895                                                     dtrain.handle))
    896         else:
    897             pred = self.predict(dtrain)

/home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/core.pyc in _check_call(ret)
    128     """
    129     if ret != 0:
--> 130         raise XGBoostError(_LIB.XGBGetLastError())
    131 
    132 

XGBoostError: [16:47:11] src/metric/metric.cc:21: Unknown metric function m

Stack trace returned 10 entries:
[bt] (0) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/./lib/libxgboost.so(_ZN7xgboost6Metric6CreateERKSs+0x5b4) [0x7fc41100b344]
[bt] (1) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/./lib/libxgboost.so(_ZN7xgboost11LearnerImpl9ConfigureERKSt6vectorISt4pairISsSsESaIS3_EE+0x733) [0x7fc410fdffb3]
[bt] (2) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/site-packages/xgboost/./lib/libxgboost.so(XGBoosterUpdateOneIter+0x5e) [0x7fc411148b8e]
[bt] (3) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/lib-dynload/_ctypes.so(ffi_call_unix64+0x4c) [0x7fc4cd5ae57c]
[bt] (4) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/lib-dynload/_ctypes.so(ffi_call+0x1f5) [0x7fc4cd5adcd5]
[bt] (5) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/lib-dynload/_ctypes.so(_ctypes_callproc+0x3e6) [0x7fc4cd5a5376]
[bt] (6) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/lib/python2.7/lib-dynload/_ctypes.so(+0x9db3) [0x7fc4cd59cdb3]
[bt] (7) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/bin/../lib/libpython2.7.so.1.0(PyObject_Call+0x53) [0x7fc4d8828e93]
[bt] (8) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/bin/../lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x715d) [0x7fc4d88db80d]
[bt] (9) /home/works/dl-tools/miniconda2/envs/tfenv.v1.2/bin/../lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x89e) [0x7fc4d88ddc3e]

```
but when I annotation this line 
```
from __future__ import unicode_literals
```
it works！
So, if there is some problem with using `unicode_literals` in my code ? Maybe the buttom C++ code should be fixed to support `unicode_literals` ?
